src/core/highlight.js

```diff
@@ -45,32 +45,61 @@ export async function run(conf) {
         console.error("Timed-out waiting for highlight:", element);
         done();
       }, 4000);
-      const msg = {
-        action: "highlight",
-        code: element.textContent,
-        id: `highlight:${i}`,
-        languages: getLanguageHint(element.classList),
-      };
-      worker.addEventListener("message", function listener(ev) {
-        const {
-          data: { id, language, value },
-        } = ev;
-        if (id !== msg.id) {
-          return; // not for us!
-        }
-        element.innerHTML = value;
-        if (element.localName === "pre") {
-          element.classList.add("hljs");
-        }
-        if (language) {
-          element.classList.add(language);
+      const codeElements = element.querySelectorAll("code");
+      if(!codeElements.length) {
```

Comment:
Note all the code duplication here. To address that have an “elementsToHighlight” array, and to it add either the pre or, if there are some, the code elements.

Reply:
Modified the above

Reply:
so:
```JS
const elementsToHighlight = element.querySelector("code") ? [...element.querySelectorAll("code")] : [element] 
```